app/models/user.rb

```diff
@@ -1,10 +1,22 @@
 class User < ApplicationRecord
-  before_save{email.downcase!}
-  validates :name,  presence: true, length: {maximum: 50}
+  attr_accessor :remember_token
+  before_save{self.email = email.downcase}
```

Comment:
what is the difference between  email.downcase and  email.downcase! ?

Reply:
downcase! sẽ trả về nil nếu như xâu email không có thay đổi giữa các kí tự.
"DownCase", "dOwncAse" => "downcase"
"downcase", "DOWNCASE" => nil

Reply:
downcase! sẽ trả về nil nếu như xâu email không có thay đổi giữa các kí tự.
"DownCase", "dOwncAse" => "downcase"
"downcase"  => nil

Reply:
How does the root string change?

Reply:
Với downcase thì trả về 1 string, còn downcase! thì trả về 1 string hoặc nil.

Reply:
I want know about the root string if you use downcase and downcase!

Reply:
Khi sử dụng downcase! với string "DOWNcase" thì trả về "downcase", root string thay đổi thành "downcase".
Với downcase, string này trả về "downcase" nhưng root string vẫn là "DOWNcase".